x-pack/platform/packages/shared/kbn-mcp-connector-common/mcp/src/schema.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { z } from '@kbn/zod';
+
+/**
+ * Schema for MCP connector HTTP service configuration.
+ *
+ * Contains only non-sensitive metadata. Credentials are validated separately
+ * via MCPConnectorSecretsSchema.
+ */
+export const MCPConnectorHTTPServiceConfigSchema = z.object({
+  http: z.object({
+    url: z.string(),
+  }),
```

Comment:
Let's keep this flat


```suggestion
  serverUrl: z.string(),
```